src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -120,6 +120,15 @@ public abstract partial class InMemoryDocumentSessionOperations : IDisposable
         protected internal readonly Dictionary<object, DocumentInfo> DocumentsByEntity =
             new Dictionary<object, DocumentInfo>(ObjectReferenceEqualityComparer<object>.Default);
 
+        /// <summary>
+        /// hold the data required to manage Counters tracking for RavenDB's Unit of Work
+        /// </summary>
+        protected internal readonly Dictionary<string, CountersCache> CountersByDocId =
+            new Dictionary<string, CountersCache> (StringComparer.OrdinalIgnoreCase);
+
+        protected internal readonly HashSet<(string docId, string counter)> CountersToDelete = new HashSet<(string docId, string Name)>();
+        protected internal readonly HashSet<(string docId, string counter)> CountersToIncrement = new HashSet<(string docId, string Name)>();
```

Comment:
What is this used for, and where is the actual amount to increment?

Reply:
Can we lazily allocate all of these? In many cases, they will not be used, and it will be extra garbage.

Reply:
removed both